brick_breaker/step_10/lib/src/components/ball.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flutter/material.dart';
+
+import '../brick_breaker.dart';
+import 'bat.dart';
+import 'brick.dart';
+import 'play_area.dart';
+
+class Ball extends CircleComponent
+    with CollisionCallbacks, HasGameReference<BrickBreaker> {
+  Ball({
+    required this.velocity,
+    required super.position,
+    required double radius,
+    required this.batWidth,
+    required this.difficultyModifier,
+  }) : super(
+            radius: radius,
+            anchor: Anchor.center,
+            paint: Paint()
+              ..color = const Color(0xff1e6091)
+              ..style = PaintingStyle.fill,
+            children: [CircleHitbox(radius: radius)]);
+
+  final Vector2 velocity;
+  final double batWidth;
+  final double difficultyModifier;
+
+  @override
+  void update(double dt) {
+    super.update(dt);
+    position += velocity * dt;
+  }
+
+  @override
+  void onCollisionStart(
+      Set<Vector2> intersectionPoints, PositionComponent other) {
+    super.onCollisionStart(intersectionPoints, other);
+    if (other is PlayArea) {
+      if (intersectionPoints.first.y <= 0) {
+        if (velocity.y < 0) velocity.y = -velocity.y;
+      } else if (intersectionPoints.first.x <= 0) {
+        if (velocity.x < 0) velocity.x = -velocity.x;
+      } else if (intersectionPoints.first.x >= game.width) {
+        if (velocity.x > 0) velocity.x = -velocity.x;
+      } else if (intersectionPoints.first.y >= game.height) {
+        add(RemoveEffect(
+            delay: 0.35,
+            onComplete: () {
+              game.playState = PlayState.gameOver;
+            }));
+      }
+    } else if (other is Bat) {
+      if (velocity.y > 0) {
+        velocity.y = -velocity.y;
+        velocity.x = velocity.x +
+            (position.x - other.position.x) / batWidth * game.width * 0.3;
```

Comment:
```suggestion
            (position.x - other.position.x) / batWidth * game.size.x * 0.3;
```
This is where I confused you I think, sorry about that.

Reply:
All good, iterating in code is great for figuring out what I don't know =)